docusaurus/docs/ui-extensions/create-an-extension-point.md

```diff
@@ -15,40 +15,271 @@ keywords:
+
+##### `options?.limitPerPlugin` - _number (Optional)_
+
+It can be used to limit maximum how many extensions should be returned from the same plugin. It can be useful in cases when there is limited space on the UI to display extensions.
+
+#### Return value
+
+The hooks return an object in the following format:
+
+```tsx
+// usePluginExtensions()
+{
+  isLoading: boolean;
+  extensions: Array<PluginExtensionLink | PluginExtensionComponent>
+}
+
+// usePluginLinkExtensions()
+{
+  isLoading: boolean;
+  extensions: PluginExtensionLink[]
+}
+
+// usePluginComponentExtensions()
+{
+  isLoading: boolean;
+  extensions: PluginExtensionComponent[]
+}
+```
+
+(For more information check the type definitions of [`PluginExtensionLink`](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/pluginExtensions.ts#L27) and [`PluginExtensionComponent`](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/pluginExtensions.ts#L35C13-L35C37).)
+
+#### Example - rendering link extensions (static context)
+
+In the following example we render a link component all link-type extensions that other plugins registered for the `plugins/another-app-plugin/menu` extension point ID.
+
+```tsx
+import { usePluginLinkExtensions } from '@grafana/runtime';
+
+// We are defining the `context` outside of the React component for performance reasons.
+// (In case we would declare it inside the component below, then it would result in a new object on every render,
+// which would unnecessarily trigger the `usePluginLinkExtensions()` hook.)
+const context = {
+  referenceId: '12345',
+  timeZone: 'UTC',
+};
+
+function AppMenuExtensionPoint() {
+  // This only returns type="link" extensions
+  const { extensions } = usePluginLinkExtensions({
+    extensionPointId: 'plugins/another-app-plugin/menu',
+    context,
+  });
+
+  if (extensions.length === 0) {
+    return null;
+  }
+
+  return (
+    <div>
+      {extensions.map((extension) => {
+        return (
+          <a href={extension.path} onClick={extension.onClick} title={extension.description} key={extension.key}>
+            {extension.title}
+          </a>
+        );
+      })}
+    </div>
+  );
+}
+```
+
+#### Example - rendering link extensions (dynamic context)
+
+In this example we are creating the context object dynamically. Although this is a common practice, we have to be aware, that the `usePluginLinkExtensions()` hook will rerender in the following scenarios:
+
+- IF the `context` object changes _(so the extensions can react to the context changes)_
+- IF the extension-registry changes
+
+Because of these we have to make sure, that we only change the `context` object if its content changes, otherwise we will cause unnecessary rerenders. The following example shows how to approach these scenarios in a safe way:
```

Comment:
```suggestion
Be sure to only change the `context` object if its content changes; otherwise, you could create unnecessary re-renders. The following example shows how to approach these scenarios in a safe way:
```